tests/unit/s2n_psk_key_exchange_modes_extension_test.c

```diff
@@ -221,6 +221,46 @@ int main(int argc, char **argv)
         };
     };
 
+    /* Test: s2n_psk_key_exchange_modes_should_send */
+    {
+        /* when neither resumption nor PSKs are enabled, the extension should not be sent */
+        {
+            DEFER_CLEANUP(struct s2n_config *no_resumption_config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(no_resumption_config);
+
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT), s2n_connection_ptr_free);
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, no_resumption_config));
+
+            EXPECT_FALSE(s2n_psk_key_exchange_modes_extension.should_send(conn));
+        };
+
+        /* when session resumption is enabled, the extension should be sent */
+        {
+            DEFER_CLEANUP(struct s2n_config *resumption_config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(resumption_config);
+            EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(resumption_config, true));
+
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT), s2n_connection_ptr_free);
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, resumption_config));
+
+            EXPECT_TRUE(s2n_psk_key_exchange_modes_extension.should_send(conn));
+        };
+
+        /* when a client is using out-of-band PSKs, the extension should be sent */
+        {
+            DEFER_CLEANUP(struct s2n_config *psk_config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(psk_config);
+
```

Comment:
You don't modify this config, so it's not actually necessary. New connections get a default config.

Same for no_resumption_config above, but I can see the argument for that one to show the difference between it and resumption_config. Still unnecessary tho, if you do want to delete it too.

Reply:
I don't like relying on default, but if you want me to delete it I can.

Reply:
Why don't you like relying on default? It should be equivalent to calling s2n_config_new().

Reply:
Like if you're worried about the default config enabling session resumption by default, s2n_config_new() could do the same ;)

Reply:
> Why don't you like relying on default?

Discussed offline. General discomfort with the default aspects of s2n-tls, which I should spend some more time learning about.

Updating PR to explicitly assert set session resumption status for all configs.